pagopa/io-app
# chore(IT Wallet): [SIW-1702] Add automatic subscription to the ITW trial system

> [!WARNING] 
> Can be tested with `io-dev-api-server` by pointing to [this PR.](https://github.com/pagopa/io-dev-api-server/pull/414)

## Short description
This PR adds an automatic subscription to the ITW trial system when opening the app.

## List of changes proposed in this pull request
- Add the `handleTrialSystemSubscription` function to check the status of the trial system. If a citizen is unsubscribed, the function should automatically subscribe them to the trial system;
- Mock the `CONFIG` object from `react-native-config`; 
- Add tests for `handleTrialSystemSubscription`.

## How to test
This PR can be tested with the `io-dev-api-server` by checking the network traffic of the app:
- Start `io-dev-api-server` with the default config which is `SUBSCRIBED`;
- Start the app with the `.env.local` config; 
- Login and check for the network traffic. This can be checked either with ProxyMan or via the logs of `io-dev-api-server`; 
- Do the same test by rotating the trial states in the configuration of `io-dev-api-server`, either by changing it directly in `config.ts` or with a custom `config.json` file. The expected behavior is: 
```
Any other state -> Only a `GET` to `trialId/subscriptions` should be performed
UNSUBSCRIBED -> Both a `GET` and a `POST` to `trialId/subscriptions` should be performed
```
